huaweicloud/services/iotda/resource_huaweicloud_iotda_batchtask.go

```diff
@@ -0,0 +1,420 @@
+		}
+
+		targetFileId = uploadTaskFileResp.FileId
+	}
+
+	createOpts := buildBatchTaskCreateParams(d, targetFileId)
+	createTaskResp, createTaskErr := client.CreateBatchTask(createOpts)
+	if createTaskErr != nil {
+		// When creating the batch task fails, it is necessary to delete the uploaded task file.
+		deleteTaskFileErr := deleteBatchTaskFile(client, *targetFileId)
+
+		return diag.Errorf("error creating IoTDA batch task: %s, %s", createTaskErr, deleteTaskFileErr)
+	}
+
+	if createTaskResp == nil || createTaskResp.TaskId == nil {
+		deleteTaskFileErr := deleteBatchTaskFile(client, *targetFileId)
+
+		return diag.Errorf("error creating IoTDA batch task: ID is not found in API response, %s", deleteTaskFileErr)
+	}
+
+	d.SetId(*createTaskResp.TaskId)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"PENDING"},
+		Target:       []string{"COMPLETED"},
+		Refresh:      batchTaskStateRefreshFunc(client, d.Id()),
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		Delay:        10 * time.Second,
+		PollInterval: 10 * time.Second,
+	}
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.Errorf("error waiting for IoTDA batch task to complete: %s", err)
+	}
+
+	return resourceBatchTaskRead(ctx, d, meta)
+}
+
+func buildBatchTaskFileUploadParams(file *os.File) *model.UploadBatchTaskFileRequest {
+	req := model.UploadBatchTaskFileRequest{
+		Body: &model.UploadBatchTaskFileRequestBody{
+			File: &def.FilePart{
+				Content: file,
+			},
+		},
+	}
+	return &req
+}
+
+func buildBatchTaskCreateParams(d *schema.ResourceData, fileId *string) *model.CreateBatchTaskRequest {
+	req := model.CreateBatchTaskRequest{
+		Body: &model.CreateBatchTask{
+			TaskName:       d.Get("name").(string),
+			TaskType:       d.Get("type").(string),
+			AppId:          utils.StringIgnoreEmpty(d.Get("space_id").(string)),
+			Targets:        utils.ExpandToStringListPointer(d.Get("targets").([]interface{})),
+			TargetsFilter:  utils.RemoveNil(d.Get("targets_filter").(map[string]interface{})),
+			DocumentSource: fileId,
+		},
+	}
+	return &req
+}
+
+func deleteBatchTaskFile(client *iotdav5.IoTDAClient, fileId string) error {
+	deleteOpts := &model.DeleteBatchTaskFileRequest{
+		FileId: fileId,
+	}
+	_, err := client.DeleteBatchTaskFile(deleteOpts)
+	if err != nil {
+		return fmt.Errorf("error deleting IoTDA batch task file after failed creation of IoTDA batch task: %s", err)
+	}
+
+	return nil
+}
+
+func batchTaskStateRefreshFunc(client *iotdav5.IoTDAClient, taskId string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		// There is no need to handle pagination related parameters, as it only applies to the structure of subtasks.
+		getOpts := &model.ShowBatchTaskRequest{
+			TaskId: taskId,
+		}
+		resp, err := client.ShowBatchTask(getOpts)
+		if err != nil {
+			if _, ok := err.(golangsdk.ErrDefault404); ok {
+				return nil, "COMPLETED", err
```

Comment:
```suggestion
				return "success", "COMPLETED", err
```

Reply:
There is no need to return "success" here